ayon_server/settings/overrides.py

```diff
@@ -169,42 +169,66 @@ def extract_overrides(
     explicit_pins: list[list[str]] | None = None,
     explicit_unpins: list[list[str]] | None = None,
 ) -> dict[str, Any]:
+    existing_overrides = {**existing} if existing else {}
+
     result: dict[str, Any] = {}
 
-    existing_overrides = {**existing} if existing else {}
-    new_overrides = paths_to_dict(explicit_pins or [])
-
-    # apply new overrides to forced ones
-    for key, value in new_overrides.items():
-        current = existing_overrides
-        if key not in current:
-            current[key] = {}
-        current = current[key]
-        if isinstance(current, dict):
-            current.update(value)  # wtf?
-
-    # remove explicit unpins
-    for path in explicit_unpins or []:
-        current = existing_overrides
-        for key in path:
-            if key not in current:
-                break
-            if key == path[-1]:
-                del current[key]
-                break
-            current = current[key]
+    print()
+    print()
+    print()
 
-    def crawl(obj, ovr, ex, target):
-        for name, _field in obj.__fields__.items():
-            child = getattr(obj, name)
-            if isinstance(child, BaseSettingsModel) and not child._isGroup:
-                if child.dict() != ovr.dict()[name] or (name in ex):
-                    target[name] = {}
-                    crawl(child, getattr(ovr, name), ex.get(name, {}), target[name])
+    def crawl(
+        original_object: BaseSettingsModel,
+        new_object: BaseSettingsModel,
+        existing_overrides: dict[str, Any],
+        target: dict[str, Any],
+        path: list[str],
+    ):
+        for field_name in original_object.__fields__.keys():
+            old_child = getattr(original_object, field_name)
+            new_child = getattr(new_object, field_name)
+            rpath = [*path, field_name]
```

Comment:
What's `rpath` variable supposed to mean? `right_path`, `real_path`? `relative_path`? `recursive_path`? Can we get a better name for it so the code is easier to grasp?

Reply:
field_path it is